torch/fx/experimental/proxy_tensor.py

```diff
@@ -194,6 +194,18 @@ def wrap_with_proxy(e, proxy, constant):
             # example use case: allreduce_ returns ([tensor], work)
             for idx, ee in enumerate(e):
                 wrap_with_proxy(ee, proxy[idx], get_constant(idx))
+        elif isinstance(e, dict):
+            assert constant is None
```

Comment:
Maybe worth a comment: in theory we could support const-prop when proxy-tensor-tracing operators that returns dicts of tensors, but we have no use case for it today (since the only op we currently trace that can return a dict is `triton_kernel_wrapper_functional/mutation`, which does not participate in const-prop)